### ISSUE TITLE
NixOS package missing version and git commit

### ISSUE THREAD

**Issue by abiosoft** [abiosoft/colima#304]
### Describe the Issue

Running `colima version` on nixpkg shows wrong version and git commit.

### Version

N/A

#### Colima Version:

What is the output of `colima version`

```
colima version development
git commit: unknown
```

#### Lima Version:

What is the output of `limactl --version`

N/A

#### Qemu Version

What is the output of `qemu-img --version`

N/A

### Operating System

- [x] macOS Intel
- [x] macOS m1
- [x] Linux

### To Reproduce

1. Install Colima via nixpkgs
2. Run `colima version`

### Expected behavior

Colima shows version and revision

### Additional context

cc @aaschmid @Atemu

**Comment by abiosoft** [comment]
@aaschmid re https://github.com/abiosoft/colima/issues/301#issuecomment-1133915369, are there any file permission restriction on the environment? That is among the likely causes.

The test is basically testing a background daemon and creates a pid file in the process. I was trying to mimic an actual use as much as possible.

Anyways, I am already looking at stubbing it out as it seems to be an issue here.

**Comment by aaschmid** [comment]
The PR https://github.com/NixOS/nixpkgs/pull/174013 on `nixpkgs` does also have failed build, see e.g. https://github.com/NixOS/nixpkgs/pull/174013/checks?check_run_id=6545228494.

**Comment by aaschmid** [comment]
I am not aware of any special permissions, though.

**Comment by aaschmid** [comment]
As a workaround we could just disable the tests for now... (ugly but effective to get the new version available).

**Comment by abiosoft** [comment]
That is fine by me. 

But how easy would it be to re-enable it? I would definitely be looking to get it fixed before the next release.

**Comment by abiosoft** [comment]
I made a new change that should simply skip the test on failure. You can try the branch once again and see.
If it passes this time, I can push a new release just for this. If otherwise, let us skip the tests for this version.

Thanks.

**Comment by Atemu** [comment]
This issue seems to be about the upgrade to the newest version. The missing version is already fixed by https://github.com/NixOS/nixpkgs/pull/173914.

It's better to track these kinds of issues in Nixpkgs though. This isn't related to this app's code, just our packaging of it.

**Comment by abiosoft** [comment]
I finally reproduced this error https://github.com/abiosoft/colima/issues/301#issuecomment-1133871253. It required me setting up Nix and running in a pure nix shell `nix-shell --pure` with this [shell.nix](https://github.com/abiosoft/colima/blob/48f89acd2a58982d216d02b5fee4077650fe6664/shell.nix).

Not a big issue and tests can be skipped for v0.4.2. It will be properly addressed ahead of future releases.

PS: current master should no longer fail the tests.

**Comment by aaschmid** [comment]
> PS: current master should no longer fail the tests.
You mean `main` from 4 days ago?

**Comment by abiosoft** [comment]
> You mean `main` from 4 days ago?

Oops yeah, I meant `main`.

**Comment by aaschmid** [comment]
Test seems to be skipped but now on my machine the following fails:
```
=== RUN   Test_config_Mounts
=== RUN   Test_config_Mounts/0
time="2022-05-29T18:56:33Z" level=fatal msg="cannot make required directory: mkdir /homeless-shelter: read-only file system"
FAIL    github.com/abiosoft/colima/environment/vm/lima  0.264s
FAIL
```

Where is the directory try to be created?

P.S.: the output for the skipped test
```
ok      github.com/abiosoft/colima/cmd  0.327s
=== RUN   TestStart
    daemon_test.go:31: pidfile /private/tmp/nix-build-colima-0.4.2.drv-0/TestStart2548514758/001/daemon.pid
    daemon_test.go:46: start successful
    daemon_test.go:53: daemon not supported: context deadline exceeded
--- SKIP: TestStart (5.01s)
PASS
```

**Comment by abiosoft** [comment]
This has been fixed for 0.3.4. after upgrading to Nix 2.8.1.

```
$ nix --version
nix (Nix) 2.8.1

$ nix shell nixpkgs#colima

$ colima version
colima version 0.3.4
git commit: 5a4a70481ca8d1e794677f22524e3c1b79a9b4ae

$ which colima
/nix/store/nygq9gxi21b939dyvkdg2w0f5xix7gng-colima-0.3.4/bin/colima
```

@aaschmid should we close the issue and track the failing tests on nixpkgs repo?

**Comment by abiosoft** [comment]
@aaschmid as a side node, what resources do I use to keep up with Nix?

I gave Nix a second try due to this issue (tried it before, gave up and reverted to Brew) and I think I finally got a hang of it. Only to check out flakes which led me to 2.0 commands, and now I am totally lost.

My use case was generally covered with `nix-shell`, `nix-shell -p ...` and home-manager. But now there's a million and one new commands that sounds like they all do the same thing. 😕

I keep seeing flakes everywhere and sounds like it is the direction Nix is heading, but there is not as much documentation compared to what I am used to with the tools I generally use. And much of docs assume you're already a Nix expert.

Anyways, enough of the rant. Closing the issue.
